### DISCUSSION TITLE
Improve error messages

### DISCUSSION THREAD

**Discussion by pierrec** [maximecb/uvm#38] [Q&A]
Having the position in the asm file where an error occurs, would help when debugging large files.

e.g.
```
thread 'main' panicked at src/vm.rs:593:13:
attempting to access data of type u64 at unaligned address
```
does not tell me where from the asm it failed and makes it hard to figure it out when the file has 12k+ lines :)

**Comment by maximecb** [comment]
Hi Pierre. I agree it would be useful to have better error messages. Did you write 12K lines of asm by hand? 😮 

To know the line in the input where the error came from, we would need to keep a mapping from input files in the asm file to positions in executable memory (`.code`).

One thing that may be a little bit helpful in the mean time is that if you run UVM with `RUST_BACKTRACE=1`, e.g.

```
RUST_BACKTRACE=1 cargo run my_program.asm
```

You should at least be able to see which instruction is causing the error.

**Reply by pierrec** [reply]
Ah no it was generated :)
Yes I can see what instruction caused it (a badly generated address that gets LOAD'ed) but unfortunately it does not happen all the time, so I need to dig into which LOAD is invalid. Will do that next.
